sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -45,12 +76,26 @@ interface DefaultCredentialConstructor {
  */
 export class DefaultManagedIdentityCredential extends ManagedIdentityCredential {
   constructor(options?: DefaultAzureCredentialOptions) {
-    const managedIdentityClientId = options?.managedIdentityClientId ?? process.env.AZURE_CLIENT_ID;
-    if (managedIdentityClientId !== undefined) {
-      super(managedIdentityClientId, options);
-    } else {
-      super(options);
+    const managedIdentityClientId =
+      (options as DefaultAzureCredentialClientIdOptions)?.managedIdentityClientId ??
+      process.env.AZURE_CLIENT_ID;
+    const managedResourceId = (options as DefaultAzureCredentialResourceIdOptions)
+      ?.managedIdentityResourceId;
+
+    // For JavaScript users.
+    if (managedIdentityClientId && managedResourceId) {
+      throw new Error(
+        `${ManagedIdentityCredential.name} - Client Id and Resource Id can't be provided at the same time.`
+      );
```

Comment:
I’m thinking I should remove this, since we’re already throwing in the ManagedIdentityCredential’s constructor if the same issue occurs.